### PR TITLE
[Android] Suppress Mobile “Switch to App” Annoyances

### PR DESCRIPTION
**Pull request by sujitacharya2005**
Resolves https://github.com/brave/brave-browser/issues/25079

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A
  
## Test Plan:
https://user-images.githubusercontent.com/32419898/216681102-b63fefaf-b1c7-4ed2-b230-afe9ad9b6485.mp4
https://user-images.githubusercontent.com/32419898/216814698-611a647b-830a-45f7-88fc-508ffccc1c91.mp4

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/shields/FilterListServiceFactory.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/shields/FilterListServiceFactory.java:4]: please use correct header, 2023 and https

File: browser/brave_shields/android/filter_list_service_factory_android.cc [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
```
Comment (SergeyZhukovsky) [inline browser/brave_shields/android/filter_list_service_factory_android.cc:4]: same about the header

File: android/java/org/chromium/chrome/browser/shields/FilterListServiceFactory.java (hunk 1 above)
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/shields/FilterListServiceFactory.java]: updated

File: browser/brave_shields/android/filter_list_service_factory_android.cc (hunk 2 above)
Comment (sujitacharya2005) [reply browser/brave_shields/android/filter_list_service_factory_android.cc]: updated

**Review by SergeyZhukovsky — APPROVED** [review]
++

File: android/java/org/chromium/chrome/browser/privacy/settings/BravePrivacySettings.java [hunk 3]
```diff
@@ -15,7 +15,10 @@
 
 import org.chromium.base.BraveFeatureList;
 import org.chromium.base.ContextUtils;
+import org.chromium.base.Log;
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/privacy/settings/BravePrivacySettings.java:18]: i don't think we use Log anywhere. so we need to remove unused imports

File: android/java/org/chromium/chrome/browser/shields/FilterListServiceFactory.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        if (nativeHandle == -1) {
+            return null;
+        }
+        MessagePipeHandle handle = wrapNativeHandle(nativeHandle);
+        FilterListAndroidHandler cookieListOptInPageAndroidHandler =
+                FilterListAndroidHandler.MANAGER.attachProxy(handle, 0);
+        Handler handler = ((Interface.Proxy) cookieListOptInPageAndroidHandler).getProxyHandler();
+        handler.setErrorHandler(connectionErrorHandler);
+
+        return cookieListOptInPageAndroidHandler;
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/shields/FilterListServiceFactory.java:52]: cookieListOptInPageAndroidHandler should be filterListAndroidHandler.

File: browser/brave_shields/android/filter_list_service_factory_android.cc [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+  auto pending =
+      brave_shields::FilterListServiceFactory::GetInstance()->GetForContext(
+          profile);
+
+  return static_cast<jint>(pending.PassPipe().release().value());
+}
+
+}  // namespace android
+}  // namespace chrome
```
Comment (deeppandya) [inline browser/brave_shields/android/filter_list_service_factory_android.cc:30]: nit : new line

File: browser/brave_shields/android/filter_list_service_factory_android.cc [hunk 6]
```diff
@@ -0,0 +1,30 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+#include "base/android/jni_android.h"
```
Comment (deeppandya) [inline browser/brave_shields/android/filter_list_service_factory_android.cc:5]: nit : new line after license string

File: browser/brave_shields/filter_list_service_factory.cc [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  auto* ad_block_service = g_brave_browser_process->ad_block_service();
+  return new FilterListService(ad_block_service);
+}
+
+content::BrowserContext* FilterListServiceFactory::GetBrowserContextToUse(
+    content::BrowserContext* context) const {
+  return chrome::GetBrowserContextRedirectedInIncognito(context);
+}
+
+}  // namespace brave_shields
```
Comment (deeppandya) [inline browser/brave_shields/filter_list_service_factory.cc:67]: nit : new line

File: browser/brave_shields/filter_list_service_factory.h [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+  KeyedService* BuildServiceInstanceFor(
+      content::BrowserContext* context) const override;
+  content::BrowserContext* GetBrowserContextToUse(
+      content::BrowserContext* context) const override;
+};
+
+}  // namespace brave_shields
+
+#endif  // BRAVE_BROWSER_BRAVE_SHIELDS_FILTER_LIST_SERVICE_FACTORY_H_
```
Comment (deeppandya) [inline browser/brave_shields/filter_list_service_factory.h:48]: nit : new line

File: components/brave_shields/browser/filter_list_service.cc [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+          filterListUuid));
+}
+
+void FilterListService::EnableFilter(const std::string& filterListUuid,
+                                     bool shouldEnableFilter) {
+  ad_block_service_->regional_service_manager()->EnableFilterList(
+      filterListUuid, shouldEnableFilter);
+}
+
+}  // namespace brave_shields
```
Comment (deeppandya) [inline components/brave_shields/browser/filter_list_service.cc:52]: nit : new line

File: components/brave_shields/browser/filter_list_service.h [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+  mojo::ReceiverSet<mojom::FilterListAndroidHandler> receivers_;
+  base::WeakPtrFactory<FilterListService> discovery_weak_factory_{this};
+
+  FilterListService(const FilterListService&) = delete;
+  FilterListService& operator=(const FilterListService&) = delete;
+};
+
+}  // namespace brave_shields
+
+#endif  // BRAVE_COMPONENTS_BRAVE_SHIELDS_BROWSER_FILTER_LIST_SERVICE_H_
```
Comment (deeppandya) [inline components/brave_shields/browser/filter_list_service.h:48]: nit : new line

**Review by deeppandya — CHANGES_REQUESTED** [review]

File: components/brave_shields/browser/filter_list_service.h (hunk 10 above)
Comment (sujitacharya2005) [reply components/brave_shields/browser/filter_list_service.h:47]: updated

File: components/brave_shields/browser/filter_list_service.cc (hunk 9 above)
Comment (sujitacharya2005) [reply components/brave_shields/browser/filter_list_service.cc:53]: updated

File: browser/brave_shields/filter_list_service_factory.h (hunk 8 above)
Comment (sujitacharya2005) [reply browser/brave_shields/filter_list_service_factory.h:48]: updated

File: browser/brave_shields/filter_list_service_factory.cc (hunk 7 above)
Comment (sujitacharya2005) [reply browser/brave_shields/filter_list_service_factory.cc:67]: updated

File: browser/brave_shields/android/filter_list_service_factory_android.cc (hunk 6 above)
Comment (sujitacharya2005) [reply browser/brave_shields/android/filter_list_service_factory_android.cc:6]: updated

File: browser/brave_shields/android/filter_list_service_factory_android.cc (hunk 5 above)
Comment (sujitacharya2005) [reply browser/brave_shields/android/filter_list_service_factory_android.cc:31]: updated

File: android/java/org/chromium/chrome/browser/shields/FilterListServiceFactory.java (hunk 4 above)
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/shields/FilterListServiceFactory.java]: updated

File: android/java/org/chromium/chrome/browser/privacy/settings/BravePrivacySettings.java (hunk 3 above)
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/privacy/settings/BravePrivacySettings.java]: updated

**Review by deeppandya — APPROVED** [review]
LGTM

**Pull request merged**